internal/sftp/goclient.go

```diff
@@ -64,23 +64,29 @@ func (c *GoClient) Upload(ctx context.Context, src io.Reader, dest string) (int6
 	return bytes, remotePath, nil
 }
 
-// Delete removes the data from dest, a new SFTP connection is opened before
+// Delete removes the data from dest. A new SFTP connection is opened before
 // removing the file, and closed when the delete is complete.
 func (c *GoClient) Delete(ctx context.Context, dest string) error {
 	if err := c.dial(); err != nil {
 		return fmt.Errorf("SFTP: unable to dial: %w", err)
 	}
 	defer c.close()
 
-	// Delete the file
-	if err := c.sftp.Remove(dest); err != nil {
-		if os.IsNotExist(err) {
-			return fmt.Errorf("SFTP: file does not exist: %w", err)
-		} else if os.IsPermission(err) {
-			return fmt.Errorf("SFTP: insufficient permissions to delete file: %w", err)
+	// SFTP assumes that "/" is used as the directory separator. See:
+	// https://datatracker.ietf.org/doc/html/draft-ietf-secsh-filexfer-02#section-6.2
+	remotePath := strings.TrimSuffix(c.cfg.RemoteDir, "/") + "/" + dest
```

Comment:
Should we not just use os path separator?

Reply:
The comment suggests that it's a slash only, i.e. not a back-slash.